pkg/controller/containerjfr/containerjfr_controller.go

```diff
@@ -111,17 +109,67 @@ func (r *ReconcileContainerJFR) Reconcile(request reconcile.Request) (reconcile.
 	}
 
 	serviceSpecs := &resources.ServiceSpecs{}
-	url, err := r.createService(context.Background(), instance, resources.NewGrafanaService(instance))
-	if err != nil {
-		return reconcile.Result{}, err
-	}
-	serviceSpecs.GrafanaAddress = fmt.Sprintf("http://%s", url)
+	var url string
+	if !instance.Spec.Minimal {
+		url, err = r.createService(context.Background(), instance, resources.NewGrafanaService(instance))
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		serviceSpecs.GrafanaAddress = fmt.Sprintf("http://%s", url)
 
-	url, err = r.createService(context.Background(), instance, resources.NewJfrDatasourceService(instance))
-	if err != nil {
-		return reconcile.Result{}, err
+		url, err = r.createService(context.Background(), instance, resources.NewJfrDatasourceService(instance))
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		serviceSpecs.DatasourceAddress = fmt.Sprintf("http://%s", url)
+
+		// check for existing minimal pod and delete if found
+		pod := resources.NewPodForCR(instance, serviceSpecs)
+		err = r.client.Get(context.Background(), types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, pod)
+		if err == nil && len(pod.Spec.Containers) == 1 {
+			reqLogger.Info("Deleting existing minimal pod")
+			err = r.client.Delete(context.Background(), pod)
+			if err != nil && !errors.IsNotFound(err) {
+				return reconcile.Result{Requeue: true, RequeueAfter: time.Second * 10}, err
```

Comment:
Just curious, what's the reason for overriding the defaults in Result here?

Reply:
Basically that creating and deleting Pods tends to be a slow process compared to everything else, so if an unexpected error results when deleting it I figured it makes sense to add a bit of delay before requeuing the re-attempt, just in case it's some transient error and the pod might actually be in the process of deletion. Without a delay then there would likely be a lot of failed re-attempts here until the pod does eventually go away.

Reply:
Understood, thanks!